align.go

```diff
@@ -205,51 +245,77 @@ func (a *Align) export(lines []string) {
 				}
 			}
 
-			word = applyPadding(word, tempColumn, a.columnCounts[columnNum], j, string(p))
+			padLength := countPadding(word, a.columnCounts[columnNum])
+			// a.padder.Grow(padLength + len(word) + (len(surroundingPad) * 2)) // TODO: might be able to do this once
+
+			paddedWord := applyPadding(a.padder, word, string(surroundingPad), tempColumn, padLength, j)
+
+			a.padder.Reset() // empty the buffer for the next iteration.
+
 			columnNum++
 			tempColumn++
 
 			// Do not add a delimiter to the last field
 			// This also properly aligns the output even if there are lines with a different number of fields
 			if a.filterLen > 0 && a.filter[a.filterLen-1] == columnNum {
-				a.writer.WriteString(word + "\n")
+				a.writer.WriteString(paddedWord + "\n")
 				break
-			} else if columnNum == len(words) {
-				a.writer.WriteString(word + "\n")
+			} else if columnNum == len(paddedWord) {
+				a.writer.WriteString(paddedWord + "\n")
 				break
 			}
-			a.writer.WriteString(word + a.sepOut)
+			a.writer.WriteString(paddedWord + a.sepOut)
 		}
 	}
 	a.writer.Flush()
 }
 
-// pad s based on the supplied PaddingOpts.
-func applyPadding(s string, columnNum, count int, j Justification, pad string) string {
-	padLength := countPadding(s, count)
+// applyPadding rebuilds word by adding padding appropriately based on the
+// desired justification, the overall padding length and the supplied surrounding
+// padding string.
+func applyPadding(padder Padder, original, surroundingPad string, columnNum, padLength int, just Justification) string {
```

Comment:
Is this function nearing the point of being a method to possibly clean up the sig all call pattern?

Reply:
I don't think I follow - sig all call?

Reply:
Auto replace wreaked havoc... Could the consideration be had to have a type that contains the contents of the arguments to this function rather than passing in all of those args on each call?

Reply:
It is certainly a fair suggestion.  I think that I would like to think on that a little further though; as that type would need to maintain the column counts and justification by column as well.